### PR TITLE
Branch enable assertion

### LINKED ISSUE #88: Assertion is not enabled in build.gradle

### PR DESCRIPTION
**Pull request by wilsonngja**
Fix #88

### CODE REVIEW

File: src/main/java/seedu/duke/parsermanager/ParseAddClient.java [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+        } catch (EmptyDetailException e) {
+            throw new EmptyDetailException(MESSAGE_ADD_CLIENT_WRONG_FORMAT);
+        } catch (InvalidEmailException e) {
+            throw new InvalidEmailException(MESSAGE_INVALID_EMAIL);
+        } catch (InvalidBudgetFormatException e) {
+            throw new InvalidBudgetFormatException(MESSAGE_INVALID_BUDGET_FORMAT);
+        }
+    }
+
+    private void checkForEmptyDetails(String commandDetail) throws EmptyDetailException {
```
Comment (ngdeqi) [inline src/main/java/seedu/duke/parsermanager/ParseAddClient.java:55]: These methods are common across all parse classes, can put these common methods in parse superclass and let the subclasses call them via super

File: src/main/java/seedu/duke/parsermanager/ParserManager.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        switch (commandType) {
+        case COMMAND_ADD:
+            ArrayList<String> processedAddCommandDetails = splitCommandAndCommandType(commandDetail);
+            String subAddCommandType = processedAddCommandDetails.get(SUB_COMMAND_INDEX);
+            String addCommandDescription = processedAddCommandDetails.get(COMMAND_DESCRIPTION_INDEX);
+
+            boolean isAddProperty = subAddCommandType.equals(PROPERTY_FLAG);
+            boolean isAddClient = subAddCommandType.equals(CLIENT_FLAG);
+
+            if (isAddProperty) {
```
Comment (ngdeqi) [inline src/main/java/seedu/duke/parsermanager/ParserManager.java:61]: can shorted to if (subAddCommandType.equals(PROPERTY_FLAG))

File: src/main/java/seedu/duke/Duke.java [hunk 3]
```diff
@@ -36,9 +36,13 @@ public void run() throws IOException {
             try {
                 //System.exit(0); //to pass CI
                 String userInputText = ui.readCommand();
-                command = parser.parseCommand(userInputText);
+
+
+                parser = parserManager.parseCommand(userInputText);
```
Comment (ngdeqi) [inline src/main/java/seedu/duke/Duke.java:41]: Maybe parseCommand can change to getParser()?

File: build.gradle [hunk 4]
```diff
@@ -43,4 +43,5 @@ checkstyle {
 
 run{
     standardInput = System.in
+    enableAssertions = true
```
Comment (ngdeqi) [inline build.gradle:46]: LGTM for this build.gradle

**Review by OVReader — APPROVED** [review]
Assertion enabled correctly in build.gradle; LGTM

**Pull request merged**